### PR TITLE
docs: add decision record for serialization contexts

### PR DESCRIPTION
**Pull request by juliapampus**
## What this PR changes/adds

Adds a decision record for issue #1581.

## Why it does that

Document thoughts and concepts.

## Further notes

PR containing the implementation will follow.

## Linked Issue(s)

Related #1581 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by paullatzelsperger — COMMENTED** [review]
To my understanding customizable serializers are intended purely for connector-to-connector communication.

For external communication (=APIs) we should continue to pursue the DTO concept.

**Comment by jimmarino** [comment]
> To my understanding customizable serializers are intended purely for connector-to-connector communication.
> 
> 
> 
> For external communication (=APIs) we should continue to pursue the DTO concept.

They should work hand-in-hand. The serializers are to control how serialization is done and configured.

**Review by ndr-brt — APPROVED** [review]

**Review by ronjaquensel — APPROVED** [review]

File: docs/developer/decision-records/2022-07-04-type-manager/README.md [hunk 1]
```diff
@@ -19,10 +20,18 @@ This should be provided by the `TypeManager`:
...
-The EDC runtime requires the ability to support configuration of multiple serialization contexts. For
-example, an API may require different serialization characteristics than another API. Related to this,
-users may want to serialize additional metadata such as JSON-LD for certain types such as extensible
-`Asset` properties or IDS messages.
+The EDC runtime requires the ability to support the configuration of multiple serialization contexts. For
+example, provided interfaces may require different serialization characteristics. Naming, some connected
+systems require JSON as input to further processing, whereas communication within an IDS ecosystem
+expect a JSON-LD string with a specific date-time-format. 
+
+To being able to adjust or replace (de-)serialization of objects to Strings and vice versa for specific
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-07-04-type-manager/README.md:28]:
```suggestion
Being able to adjust or replace (de-)serialization of objects and vice versa for specific
```

File: docs/developer/decision-records/2022-07-04-type-manager/README.md [hunk 2]
```diff
@@ -19,10 +20,18 @@ This should be provided by the `TypeManager`:
...
+extensions or added properties (e.g., caused by a domain-specific extension of an `Asset`), the `TypeManager`
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-07-04-type-manager/README.md:29]:
```suggestion
extensions or added properties (e.g. required by a domain-specific extension of an `Asset`), the `TypeManager`
```

File: docs/developer/decision-records/2022-07-04-type-manager/README.md [hunk 3]
```diff
@@ -19,10 +20,18 @@ This should be provided by the `TypeManager`:
...
+allows to register customized processors that extend or overwrite default functionalities.
+
+This feature and decision targets the adaptability to existing interfaces and systems and the interoperability
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-07-04-type-manager/README.md:32]:
```suggestion
This feature targets the adaptability to existing interfaces and systems and the interoperability
```

File: docs/developer/decision-records/2022-07-04-type-manager/README.md [hunk 4]
```diff
@@ -19,10 +20,18 @@ This should be provided by the `TypeManager`:
...
+requirements within varying data spaces. What is not intended is customization on a level that e.g. each
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-07-04-type-manager/README.md:33]:
```suggestion
requirements within different data spaces. The goal is not to enable customization on the API level, such that e.g. each
```
(each dataspace in itself does not vary...)

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**